rust/core-lib/src/tabs.rs

```diff
@@ -371,9 +372,13 @@ impl<W: Write + Send + 'static> Documents<W> {
             let buffer_id = self.next_buffer_id();
             self.new_empty_view(rpc_peer, &view_id, buffer_id);
         }
+
+        // schedule plugin notification for after the view has received an identifer
+        let view_id2 = view_id.to_owned();
         let init_info = self.buffers.lock().editor_for_view(&view_id)
             .unwrap().plugin_init_info();
-        self.plugins.document_new(&view_id, init_info);
+        self.plugins.run_on_idle(move |p| p.document_new(&view_id2, &init_info));
```

Comment:
View creation feels like a special case, because it is responsible for generating and assigning `ViewIdentifier`s, which in turn are used for all message routing.

On 1/2 I agree. 3. isn't a problem, because the frontend can't send a command until it knows what commands are available, which it can't know until it has a view id. On 4., yes, I think a good rule of thumb is that if we ever add a `run_with_delay` function we've probably screwed up somewhere. 😏

How about this solution: when a client view has received an identifier, it sends a `view_loaded` notification, and we use _that_ to do the setup?